app/ViewModels/TransactionViewModel.php

```diff
@@ -83,9 +82,8 @@ public function dateTime(): Carbon
 
     public function nonce(): int
     {
-        $wallet = Wallets::findByPublicKey($this->transaction->sender_public_key);
-
-        return $wallet->nonce->toNumber();
+        /* @phpstan-ignore-next-line */
```

Comment:
why the ignore? Also seems to fail so you may be using something else than you think here

Reply:
Really unsure 🤔 Using this I'm able to get the right nonce for the given transaction, while before it wasn't the case, tried it directly on explorer by going through transactions details and was just working fine 🤔

Reply:
What's the error that resulted in the ignore?

Reply:
Access to undefined property `nonce`, so like you told me, had to declare it on the `Transaction` model class with `@property`, so now phpstan error is gone, but when running tests I still get `  App\ViewModels\TransactionViewModel::nonce(): Return value must be of type int, null returned` so looking into that now.